src/compiler/checker.ts

```diff
@@ -12106,7 +12106,7 @@ namespace ts {
         }
 
         function getApplicableIndexInfoForName(type: Type, name: __String): IndexInfo | undefined {
-            return getApplicableIndexInfo(type, getStringLiteralType(unescapeLeadingUnderscores(name)));
+            return getApplicableIndexInfo(type, isLateBoundName(name) ? esSymbolType : getStringLiteralType(unescapeLeadingUnderscores(name)));

```

Comment:
Is a late-bound-name always an ES Symbol? @weswigham says not.

Reply:
Maybe we need to rename this to `isUnresolvedLateBoundName` or something like that.